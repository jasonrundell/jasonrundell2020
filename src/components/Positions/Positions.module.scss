.list {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-right: $normalUnit;
  margin-bottom: $normalUnit;
  list-style: none;
}

.item {
  display: flex;
  flex-direction: row;
  width: 100%;
  font-size: $smallUnit;
  justify-content: space-between;

  span {
    font-size: $smallUnit;
    flex: 1;
  }
}
