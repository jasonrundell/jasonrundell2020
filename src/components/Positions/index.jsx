import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'

// CSS
import styles from './Positions.module.scss'

const Positions = ({ data }) => {
  const allPositions = data.allContentfulPositions.edges

  return (
    <>
      <ul className={styles.list}>
        {allPositions.map(position => {
          const { id, role, company, startDate, endDate } = position.node
          return (
            <li key={id} className={styles.item}>
              <h3 className={styles.role}>{role}</h3>
              <span className={styles.company}>{company}</span>
              <br />
              <span className={styles.startDate}>{startDate}</span> -{' '}
              <span className={styles.endDate}>{endDate}</span>
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default props => (
  <StaticQuery
    query={graphql`
      {
        allContentfulPositions(sort: { fields: orderId, order: DESC }) {
          edges {
            node {
              id
              orderId
              role
              company
              startDate
              endDate
            }
          }
        }
      }
    `}
    render={data => <Positions data={data} {...props} />}
  />
)
Positions.propTypes = {
  data: PropTypes.shape({
    allContentfulPositions: PropTypes.shape({
      edges: PropTypes.arrayOf(PropTypes.object).isRequired,
    }).isRequired,
  }).isRequired,
}
