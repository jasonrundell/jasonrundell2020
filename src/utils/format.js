/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
