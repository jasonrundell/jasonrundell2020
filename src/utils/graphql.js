export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
