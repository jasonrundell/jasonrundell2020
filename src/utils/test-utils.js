export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
