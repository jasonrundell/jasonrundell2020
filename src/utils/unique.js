const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
