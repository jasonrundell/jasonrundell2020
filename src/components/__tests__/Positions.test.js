import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Positions from '../Positions'

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const testData = [
  {
    id: '21694cf0-7b34-5678-ab04-114728a8802c',
    orderId: 7,
    role: 'Business Owner',
    company: 'Code Summoner Corp.',
    startDate: 'April 2012',
    endDate: 'Present',
  },
  {
    id: 'bc304f5c-5089-534e-be51-e7754a84a045',
    orderId: 6,
    role: 'Senior Front End Web Developer',
    company: 'Intuit',
    startDate: 'July 2019',
    endDate: 'Present',
  },
]

describe('Positions component', () => {
  it('Positions renders correctly', () => {
    const tree = renderer.create(<Positions positions={testData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Positions has required positions prop', () => {
    act(() => {
      render(<Positions positions={testData} />, container)
    })
  })
})
