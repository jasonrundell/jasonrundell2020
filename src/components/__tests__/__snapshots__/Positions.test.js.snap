// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Positions component Positions renders correctly 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <h2
      className="h2"
    >
      Business Owner
    </h2>
    <span
      className="company"
    >
      Code Summoner Corp.
    </span>
    <br />
    <span
      className="startDate"
    >
      April 2012
    </span>
     -
     
    <span
      className="endDate"
    >
      Present
    </span>
  </li>
  <li
    className="item"
  >
    <h2
      className="h2"
    >
      Senior Front End Web Developer
    </h2>
    <span
      className="company"
    >
      Intuit
    </span>
    <br />
    <span
      className="startDate"
    >
      July 2019
    </span>
     -
     
    <span
      className="endDate"
    >
      Present
    </span>
  </li>
</ul>
`;
